runtime/runtime-params-estimator/src/transaction_builder.rs

```diff
@@ -95,15 +103,12 @@ impl TransactionBuilder {
         self.accounts[account_index].clone()
     }
     pub(crate) fn random_unused_account(&mut self) -> AccountId {
-        if self.used_accounts.len() == self.accounts.len() {
+        if self.unused_index == self.accounts.len() {
             panic!("All accounts used. Try running with a higher value for the parameter `--accounts-num <NUM>`.")
         }
-        loop {
-            let account = self.random_account();
-            if self.used_accounts.insert(account.clone()) {
-                return account;
-            }
-        }
+        let tmp = self.unused_index;
+        self.unused_index += 1;
+        return self.accounts[self.unused_accounts[tmp]].clone();
```

Comment:
Optional: Maybe you can simplify this by moving things around a bit. Roughly like this:
```rust
if let Some(index) = unused_indices.pop() {
    accounts[index].clone()
} else {
    panic!(...)
}
```

Reply:
The reason why I used index rather than `pop` is because I saw the [pop source code](https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#1790) has some unsafe code in it. I was not sure of its pros and cons. 

I can do similar check on index like `if index >= len(unused_accounts)` then panic.

Let me know what you think. I am okay with either options.

Reply:
I have changed the existing check in the code to `if self.unused_index >= self.unused_accounts.len()`. The change should ensure the index does not go out of bounds.  I personally prefer this approach over `pop`.

Reply:
I like that change, now it's clear from local reasoning that `self.unused_index` is in bound of `unused_accounts`.

With regard to avoiding `pop`, I can see your reasoning with unsafe code. However, in this case I wouldn't see it as a strong argument. Unsafe code isn't always bad. When dealing with primitive memory, such as `Vec` does, there is bound to be some unsafe code. The standard library provides a safe interface to it and I trust the maintainers of Rust to have no bugs in there. And in this particular case, it is not too difficult to see that it is correct, if you know a bit about the internals of `Vec`.

But anyway, your implementation looks good, I will merge it now.